Doctrine/Phpcr/SitemapUrlInformationProvider.php

```diff
@@ -0,0 +1,169 @@
+     */
+    private $router;
+
+    /**
+     * @var SecurityContextInterface
+     */
+    private $publishWorkflowChecker;
+    /**
+     * @var string
+     */
+    private $defaultChanFrequency;
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+    /**
+     * @var SeoPresentation
+     */
+    private $seoPresentation;
+
+    /**
+     * @param DocumentManager $manager
+     * @param RouterInterface $router
+     * @param string $defaultChanFrequency
+     * @param PsrLogger $logger
+     * @param SeoPresentation $seoPresentation
+     * @param SecurityContextInterface $publishWorkflowChecker
+     */
+    public function __construct(
+        DocumentManager $manager,
+        RouterInterface $router,
+        $defaultChanFrequency,
+        PsrLogger $logger,
+        SeoPresentation $seoPresentation,
+        SecurityContextInterface $publishWorkflowChecker = null
+    ) {
+        $this->manager = $manager;
+        $this->router = $router;
+        $this->defaultChanFrequency = $defaultChanFrequency;
+        $this->logger = $logger;
+        $this->seoPresentation = $seoPresentation;
+        $this->publishWorkflowChecker = $publishWorkflowChecker;
+    }
+
+    /**
+     * {@inheritDocs}
+     */
+    public function getUrlInformation()
+    {
+        $contentDocuments = $this->fetchSitemapDocuments();
+        $urlInformationList = array();
+
+        foreach ($contentDocuments as $document) {
+            if (null != $this->publishWorkflowChecker &&
+                !$this->publishWorkflowChecker->isGranted(array(PublishWorkflowChecker::VIEW_ATTRIBUTE), $document)
+            ) {
+                continue;
+            }
+
+            try {
+                $urlInformationList[] = $this->computeUrlInformationFromSitemapDocument($document);
+            } catch (\Exception $e) {
+                $this->logger->info($e->getMessage());
+            }
+        }
+
+        return $urlInformationList;
+    }
+
+    /**
+     * @param AlternateLocaleProviderInterface $alternateLocaleProvider
+     */
+    public function setAlternateLocaleProvider(AlternateLocaleProviderInterface $alternateLocaleProvider)
+    {
+        $this->alternateLocaleProvider = $alternateLocaleProvider;
+    }
+
+    /**
+     * Wrapper to fetch the sitemap documents from database.
+     *
+     * @return array []
+     *
+     * @throws \Doctrine\ODM\PHPCR\Query\QueryException
+     */
+    protected function fetchSitemapDocuments()
+    {
+        // todo rewrite query when https://github.com/symfony-cmf/CoreBundle/issues/126 is ready
+        return $this->manager->createQuery(
+            "SELECT * FROM [nt:unstructured] WHERE (visible_for_sitemap = true)",
+            QueryInterface::JCR_SQL2
+        )->execute();
+    }
+
+    /**
+     * Transforms a single sitemap document into url information.
+     *
+     * A sitemap document is a document, which should be exposed on a sitemap.
+     *
+     * @param $document
```

Comment:
`@param object $document The content document to build the url information for.`.